api.go

```diff
@@ -119,3 +130,29 @@ func main() {
 	}
 	log.Fatal(s.ListenAndServeTLS("", ""))
 }
+
+func handleDnsRequest(w dns.ResponseWriter, r *dns.Msg) {
+	if len(r.Question) > 0 && strings.HasSuffix(r.Question[0].Name, ".play-with-docker.com.") && strings.HasPrefix(r.Question[0].Name, "ip") {
+		// this is something we know about and we should try to handle
+		question := r.Question[0].Name
+		domainChunks := strings.Split(question, ".")
+		tldChunks := strings.Split(strings.TrimPrefix(domainChunks[0], "ip"), "-")
```

Comment:
Question: "-" might now always come right?. Remember that we can also handle stuff like ip10_0_0_1.play-with-docker.com:8080 for which this `split` will fail.